src/Lifecycle/Guards.php

```diff
@@ -61,7 +60,8 @@ public function validate(): static
     protected function passesAuthorization(): bool
     {
         if (method_exists($this->event, 'authorize')) {
-            $result = app()->call([$this->event, 'authorize']);
+            $resolver = DependencyResolver::for($this->event->authorize(...), event: $this->event);
+            $result = call_user_func_array([$this->event, 'authorize'], $resolver());
```

Comment:
Right now `authorize`, `fired`, and `handle` do not support prefixes (ie. `authorizeFoo` will not be called by Verbs—only `authorize`). There are reasons for this, but I wonder if we want to make it use the same prefix logic everywhere to be consistent.

Reply:
My first thought was that it could make the events hard to reason about if all the functions can have prefixed instances. `validate` and `apply` inherently make sense to me since the logic within is so specific to the states being passed - this may be a symptom of knowing the Verbs lifecycle already though.

Consistency is nice too though 🤷